### PR TITLE
Update static docs structure and information

### PR DESCRIPTION
**Pull request by herrardo**
…from development.

EX-5608

Change static docs structure to provide a better understanding to the developer user.

# Pull request template

The root document, README.md has been changed to provide two scenarios to the developer user:

- Integration
- Development

Currently, we have two sections explaining the integration by Archetype or by Components. But in both cases, we are talking about how to develop a search experience. So those two sections have been changed to use the word **_Develop_** instead.

We have added another section talking about integration to link a new document created by @javieri-empathy which is the documentation about integrating the archetype into a customer website in the same manner as we did for EmpathyX, loading the script and initialize the snippet.

I took advantage of this PR to also add information about the new feature added by @guillei10 `setSnippetConfig`and also to do a deeper explanation about callbacks and events to subscribe.

## Motivation and context
We haven't published the integration documentation in eDocs portal yet and some clients have asked about it.

- [ ] Dependencies. If any, specify:
- [x] Open issue. If applicable, link: EX-5608

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that causes existing functionality to not work as expected)
- [x] Change requires a documentation update

## What is the destination branch of this PR?

- [x] `Main`
- [ ] Other. Specify:

## How has this been tested?

Tests performed according to [testing guidelines](./contributing/tests.md): 

## Checklist:

- [x] My code follows the **[style guidelines](./CONTRIBUTING.md#style-guides)** of this project.
- [x] I have performed a **self-review** on my own code.
- [ ] I have **commented** my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the **[documentation](./CONTRIBUTING.md#documentation-style-guide)**.
- [ ] My changes generate **no new warnings**.
- [ ] I have added **tests** that prove my fix is effective or that my feature works.
- [ ] New and existing **unit tests pass locally** with my changes.

### CODE REVIEW

**Comment by herrardo** [comment]
Corrected some warning and links too

File: packages/x-components/static-docs/build-search-ui/web-archetype-integration-guide.md [hunk 1]
```diff
@@ -161,26 +161,71 @@ invoked at any time. Note that you should only call this function **once**.
...
+| Function         | Parameters                                                                     | Description                                                                                                                                                       |
+| ---------------- | ------------------------------------------------------------------------------ | ----------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| init             | - [Snippet Configuration](#snippet-configuration) - The initialisation options | [Initialises Interface X on demand](#initialise-on-demand).                                                                                                       |
+| search           | - query (Optional) - The query to open Interface&nbsp;X with                   | Opens Interface&nbsp;X with the given search query.                                                                                                               |
```
Comment (LuisMartinez15) [inline packages/x-components/static-docs/build-search-ui/web-archetype-integration-guide.md:230]:
```suggestion
| search           | - query (Optional) - The query to open Interface&nbsp;X with                   | Opens Interface&nbsp;X and triggers a search with the given query.                                                                                                               |
```

File: packages/x-components/static-docs/build-search-ui/web-archetype-integration-guide.md [hunk 2]
```diff
@@ -161,26 +161,71 @@ invoked at any time. Note that you should only call this function **once**.
...
+| setSnippetConfig | - [Snippet Configuration](#snippet-configuration) - The Initialisation options | Changes the options and all components react to the changes. Helps to change, language, for instance, to respond to user interactions without reloading the page. |
```
Comment (LuisMartinez15) [inline packages/x-components/static-docs/build-search-ui/web-archetype-integration-guide.md:231]:
```suggestion
| setSnippetConfig | - [Snippet Configuration](#snippet-configuration) - The Initialisation options | Changes the options and all components react to the changes. Useful, for example, to change both search engine and displayed language without reloading the page. |
```

**Review by LuisMartinez15 — CHANGES_REQUESTED** [review]
Good job! Doc is looking promising!

**Review by LuisMartinez15 — APPROVED** [review]
LGTM. Good job!

**Review by mlcorton — APPROVED** [review]
Great job, guys! Let's see how it shines in eDocs. Later on, I'll make a deeper rev and open a new PR if appropriate :)

**Pull request merged**